### PR TITLE
[SPARK-24061][SS]Add TypedFilter support for continuous processing

### PR DESCRIPTION
**Pull request by yanlin-Lynn**
## What changes were proposed in this pull request?

Add TypedFilter support for continuous processing application.

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by yanlin-Lynn** [comment]
@xuanyuanking , please help to review for this path. Thank you!

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala [hunk 1]
```diff
@@ -771,7 +778,16 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
...
+  /** Assert that the logical plan is not supportd for continuous procsssing mode */
+  def assertSupportedForContinuousProcessing(
+    name: String,
+    plan: LogicalPlan,
+    outputMode: OutputMode): Unit = {
+    test(s"continuous processing - $name: supported") {
+      UnsupportedOperationChecker.checkForContinuous(plan, outputMode)
+    }
+  }
+      /**
```
Comment (xuanyuanking) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: nits: Indent here

**Review by xuanyuanking — COMMENTED** [review]
Just small nits. LGTM to me

**Comment by xuanyuanking** [comment]
+1 for this.
We find this by CP app use filter with functions, this can be supported by current implement.
cc @jose-torres @zsxwing @tdas

**Comment by jose-torres** [comment]
LGTM. TypedFilter and Filter share the FilterExec execution node, so this should just work.

Ideally we would add a test to ContinuousSuite to ensure that TypedFilter does execute properly, but I understand this may not be sustainable. Eventually we'll just remove this whitelist and find some way to broadly execute all the streaming tests for continuous processing.

**Comment by HeartSaVioR** [comment]
LGTM. This is what I also found so far today.

**Comment by zsxwing** [comment]
okey to test

**Comment by zsxwing** [comment]
ok to test

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala [hunk 2]
```diff
@@ -771,6 +778,16 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
     }
   }
 
+  /** Assert that the logical plan is not supportd for continuous procsssing mode */
```
Comment (arunmahadevan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: is not supported -> is supported

**Review by arunmahadevan — COMMENTED** [review]
+1, hit the same issue.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala (hunk 2 above)
Comment (yanlin-Lynn) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: ah, you got it!

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala [hunk 3]
```diff
@@ -840,4 +857,9 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
     def this(attribute: Attribute) = this(Seq(attribute))
     override def isStreaming: Boolean = true
   }
+
+  case class StreamingRelationV2(output: Seq[Attribute]) extends LeafNode {
```
Comment (zsxwing) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: nit: rename this to `TestStreamingRelationV2`

**Review by zsxwing — CHANGES_REQUESTED** [review]
LGTM except one nit

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala [hunk 4]
```diff
@@ -771,6 +778,16 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
     }
   }
 
+  /** Assert that the logical plan is supportd for continuous procsssing mode */
```
Comment (zsxwing) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: nit:`supportd` -> `supported`

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala (hunk 3 above)
Comment (yanlin-Lynn) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: I have tried to do that, but the code in [UnsupportedOperationChecker](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala#L349) will fail the UT. So, should I change the checking logic there too?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala (hunk 4 above)
Comment (yanlin-Lynn) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: ah, my bad!. Thanks for pointing it out.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala (hunk 3 above)
Comment (yanlin-Lynn) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala]: ack

**Comment by zsxwing** [comment]
LGTM. Merging to master. Thanks!

**Comment by zsxwing** [comment]
retest this please

**Pull request closed without merging**